core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -992,6 +999,18 @@ public String call() throws IOException {
             catch (Throwable ex) { return "< 1.335"; } // Older agent.jar won't have VERSION
         }
     }
+
+    private static final class CommunicationProtocol extends MasterToSlaveCallable<String,IOException> {
+        @Override
+        public String call() throws IOException {
+            try {
+                if (Engine.current() != null)
```

Comment:
Static analysis will be unhappy unless you assign the value to a local variabe

Reply:
Oh, Thank you! I'll fix it.